src/Resolver/WishlistsResolver.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\Resolver;
+
+use BitBag\SyliusWishlistPlugin\Repository\WishlistRepositoryInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+final class WishlistsResolver implements WishlistsResolverInterface
+{
+    private WishlistRepositoryInterface $wishlistRepository;
+
+    private string $wishlistCookieToken;
+
+    private TokenStorageInterface $tokenStorage;
+
+    public function __construct(
+        WishlistRepositoryInterface $wishlistRepository,
+        string $wishlistCookieToken,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->wishlistRepository = $wishlistRepository;
+        $this->wishlistCookieToken = $wishlistCookieToken;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function resolve(Request $request): ?array
```

Comment:
Instead of passing Request it would be more convinient to pass "value" you're  looking for.